File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/ui/bike/bikeList/ScanBikeQRCodeActivityView.java
package com.lattis.ellipse.presentation.ui.bike.bikeList;

import com.lattis.ellipse.domain.model.Bike;
import com.lattis.ellipse.domain.model.Card;
import com.lattis.ellipse.domain.model.Lock;
import com.lattis.ellipse.presentation.model.LockModel;
import com.lattis.ellipse.presentation.ui.base.activity.bluetooth.BaseBluetoothActivityView;

import java.util.List;

/**
 * Created by ssd3 on 9/5/17.
 */

public interface ScanBikeQRCodeActivityView extends BaseBluetoothActivityView {

    void onGetCardSuccess(List<Card> cards);
    void onGetCardFailure();

    void onInvalidQRCode();
    void restartScanner();

     void onBikeDetailsSuccess(Bike bike);

    void onBikeUnAuthorised();
    void onBikeAlreadyRented();
    void onBikeNotFound();
    void onBikeNotAvailable();

     void onBikeDetailsFailure();

    void startRide();
    void onStartRideSuccess();
    void onStartRideFail();


    void OnReserveBikeSuccess();
    void OnReserveBikeFail();
    void OnReserveBikeNotFound();

    void onCancelBikeSuccess();
    void onCancelBikeFail();


    ////////////////////////////////// LOCK CONNECTION CODE : START //////////////////////////////////////////////////

    void onLockScanned(LockModel lockModel);
    void onScanStart();
    void onScanStop();


    void onLockConnected(LockModel lockModel);
    void showConnecting(boolean requiresReconnection);
    void onLockConnectionFailed();
    void onLockConnectionAccessDenied();


    void OnSignedMessagePublicKeySuccess(String signedMessage, String publicKey);
    void OnSignedMessagePublicKeyFailure();


    void onSaveLockSuccess(Lock lock);
    void onSaveLockFailure();

    void OnSetPositionStatus(Boolean status);

    void onGetUserProfile();

}

File: velolabs_repos/lattis_android_v2/app/src/main/java/com/lattis/lattis/presentation/qrscan/ScanBikeQRCodeActivityView.kt
package com.lattis.lattis.presentation.qrscan

import com.lattis.domain.models.Bike
import com.lattis.domain.models.Card
import com.lattis.domain.models.Lock
import com.lattis.lattis.presentation.base.activity.bluetooth.BaseBluetoothActivityView
import com.lattis.lattis.uimodel.model.LockModel

interface ScanBikeQRCodeActivityView : BaseBluetoothActivityView {
    fun handleCard(card: Card)
    fun handleNoCard()
    fun onInvalidQRCode()
    fun restartScanner()
    fun onBikeDetailsSuccess(bike: Bike)
    fun onBikeUnAuthorised()
    fun onBikeAlreadyRented()
    fun onBikeNotFound()
    fun onBikeNotAvailable()
    fun onBikeDetailsFailure()
    fun onBikeNotLive()
    fun startRide()
    fun onStartRideSuccess()
    fun onStartRideFail()
    fun OnReserveBikeSuccess()
    fun OnReserveBikeFail()
    fun OnReserveBikeNotFound()
    fun onMissingUserCard()
    fun onCancelBikeSuccess()
    fun onCancelBikeFail()
    fun hideProgressbar()
    fun showProgressbar()

    ////////////////////////////////// LOCK CONNECTION CODE : START //////////////////////////////////////////////////
    fun onLockScanned(lockModel: LockModel?)
    fun onScanStart()
    fun onScanStop()
    fun onLockConnected(lockModel: LockModel)
    fun showConnecting(requiresReconnection: Boolean)
    fun onLockConnectionFailed()
    fun onLockConnectionAccessDenied()
    fun OnSignedMessagePublicKeySuccess(
        signedMessage: String,
        publicKey: String
    )

    fun OnSignedMessagePublicKeyFailure()
    fun onSaveLockSuccess(lock: Lock?)
    fun onSaveLockFailure()
    fun OnSetPositionStatus(status: Boolean)
    fun onGetUserProfile()

    fun onCodeSentSuccess()
    fun onCodeSentFailure()
    fun onCodeValidateFailure()


    fun onIotScanSuccess()
    fun onIotScanFailure()
    fun setIotScanSubTitle()


    //// rentalfare :start
    fun showPayPerUse()
    fun showRentalFare()
    fun onBikeReserveFailureDuePricingOptionSelectionRemaining()
    //// rentalfare :end


    ////closehub :start
    fun showClosedHubPorts()
    fun onClosedHubPortSelectionFailure()
    fun onClosedHubPortSelectionSuccess()
    ////closehub :end
}